### ISSUE TITLE
1.3.0's MaybePromise is applied to parent types of Typescript resolvers

### ISSUE THREAD

**Issue by jrr** [dotansimha/graphql-code-generator#2026]
**Describe the bug**
1.3.0 applies the new `MaybePromise<T>` to the parent and return types of resolvers. It broke a bunch of our resolvers that were reading values off of their parent.

```graphql
# schema
type Employee {
  id: Int!
  name: String!
}
```

**This used to work:**
```ts
// resolver
import { EmployeeResolvers } from "./types.gen";

export const employeeResolvers: EmployeeResolvers = {
  name: async (parent, args, context) => {
    return parent.name.toUpperCase();
  }
};
```

**1.3.0 error:**

```
foo.ts:5:19 - error TS2339: Property 'name' does not exist on type 'MaybePromise<Employee>'.
  Property 'name' does not exist on type 'Promise<Employee>'.

5     return parent.name.toUpperCase();
                    ~~~~
```

I could work around it by awaiting the parent:
```ts
return (await parent).name.toUpperCase();
```

..but I'd rather not. They shouldn't be necessary, and we have about 200 of them 😬

https://github.com/dotansimha/graphql-code-generator/issues/1965 
https://github.com/dotansimha/graphql-code-generator/pull/2023

**Comment by kamilkisiela** [comment]
This one reverts to the previous behavior and hides promises behind a flag:
https://github.com/dotansimha/graphql-code-generator/pull/2044

**Comment by kamilkisiela** [comment]
Actually, we changed the behavior a bit and now it matches what was before and introduces nested promises, without breaking anything.

https://github.com/dotansimha/graphql-code-generator/pull/2044

**Comment by dotansimha** [comment]
Fixed in 1.3.1
